lib/phoenix/template.ex

```diff
@@ -330,6 +347,10 @@ defmodule Phoenix.Template do
         unquote(quoted)
       end
 
+      defp render_template(unquote(name), assigns) do
+        unquote(defp)(assigns)
```

Comment:
This is duplicating the template AST everywhere. We should make the regular render call render_template.